framework/db/Connection.php

```diff
@@ -713,12 +713,22 @@ public function transaction(callable $callback, $isolationLevel = null)
             }
         } catch (\Exception $e) {
             if ($transaction->isActive && $transaction->level === $level) {
```

Comment:
These two blocks are exactly the same. Is it possible to extract these into a method?